TA-Pintade <br/>
Ana Maria Renno Pocai <br/>
Xinyi WANG

<font color = blue>
Objectif de ce travail:
</font>

Selon le vidéo proposé sur pintade, nous avons cherché les informations et modelisé le problème (comment gagner le plus de profit par le couple de pintade).


<font color = blue>
Modélisation du problème: <br/>
</font>


<font color = green>
Partie 1: Calculer les nombre des pintades maintenant <br/>
</font>

Ces deux monsieurs ont un couple de pintades (il y a 2 ans). 

Les 6 premiers mois, les pintades ne produisent rien (Suppose que c’est dans l’automne et l’hiver). 
On suppose que le côut par pintade par mois est 5 euros.

Au septième mois, la pintade femelle commence à produire des œufs(Pendant 6 mois chaque année, au printemps et en été).

Elle arrête de produire des œufs jusqu'à  à l'âge d'un an et demi. Une pintade peut produire entre 50 et 80 œufs par an. 

Étant donné que les deux messieurs n'ont rien fait avec le couple, comme cela a été dit dans la vidéo, les pintades se reproduisent frénétiquement. Pour cet exercice, nous allons considérer qu'elles produisent 50 œufs par an, en tenant compte des facteurs externes qui pourraient diminuer le nombre de pintades.

La taux d'éclosion des pintades est d'environ 80 %.

En tenant compte de ces informations, nous pouvons estimer la quantité de pintades après 2 ans.

In [63]:
oeuf_ans = 50
eclosion = 0.8
femmes = 1/2
cout_pintada = 5 
print("Coût de l'élevage d'une pintade par mois: ", cout_pintada, "€")

Coût de l'élevage d'une pintade par mois:  5 €


#### Première année: 

En considérant que la pintade ne se reproduit que pendant 6 mois de l'année et qu'elle met 1 mois à éclore, faisons une moyenne, en supposant qu'il naîtra 8 nouvelles pintades chaque mois après le premier mois d'incubation, afin de calculer le coût des pintades pendant cette période.

Pour faciliter le calcul, considérons le cas idéal où l'agriculteur a acheté les pintades 6 mois avant leur période de reproduction. Donc, au cours des 7 premiers mois, nous n'aurons que le coût de 2 pintades.

In [64]:
pintadas_1 = int(oeuf_ans*eclosion) + 2
pintadas_femmes_1 = int(pintadas_1*femmes) 
p = 2 
cout_pintadas_1 = p*5*7 
for i in range(5):
    p += 8
    cout_pintadas_1 += p*5
print("Nombre de pintades: ", pintadas_1)
print("Nombre de femelles: ", pintadas_femmes_1)
print("Coût des pintades: ", cout_pintadas_1, "€")

Nombre de pintades:  42
Nombre de femelles:  21
Coût des pintades:  720 €


#### Deuxième année:

In [65]:
p = pintadas_1
cout_pintadas_2 = p*5*7 
for i in range(5):
    p += 8*pintadas_femmes_1
    cout_pintadas_2 += p*5
pintadas_2 = int(pintadas_femmes_1*oeuf_ans*eclosion) + pintadas_1
pintadas_femmes_2 = int(pintadas_2*femmes)
print("Nombre de pintades: ", pintadas_2)
print("Nombre de femelles: ", pintadas_femmes_2)
print("Coût des pintades: ", cout_pintadas_2, "€")

Nombre de pintades:  882
Nombre de femelles:  441
Coût des pintades:  15120 €


<font color = green>
Partie 2: Coût d'incubation <br/>
</font>


 L'incubation est faite artificiellement
Si l'incubation est faite artificiellement, la pintade peut continuer à pondre des œufs. Dans ce cas, on peut estimer qu’une pintade produit 192-240 œufs (pintades). Mais il y a le coût de l’incubation artificielle(2 euros par pintade: 384 - 480 euros).

In [66]:
cout_incubation = 2*192 # Une pintade.
print("Coût de incubation par une mère pintada: ",cout_incubation, "€"  )

Coût de incubation par une mère pintada:  384 €


<font color = green>
Partie 3: Prix <br/> 
</font>

#### Prix de vente des œufs de pintade: 

Le prix de vente de chaque œuf de pintade est basé sur le fait que 6 œufs se vendent pour 7 euros. 

Cela signifie que le prix par œuf est d'environ :

$$
\text{Prix par œuf} = \frac{7 \text{ euros}}{6 \text{ œufs}} \approx 1,17 \text{ euros}
$$

#### Calcul du Coût de Production d'une Boîte d'Oeufs

Pour déterminer le coût de production d'une boîte d'œufs (qui contient 6 œufs), nous avons pris en compte le coût de production individuel de chaque œuf. Voici comment nous avons calculé cela :

1. **Coût de production par œuf** : Nous avons mentionné que le coût de production d'un œuf est de **0,3 euros**.

2. **Coût total par boîte d'œufs** : Pour calculer le coût total d'une boîte d'œufs, nous multiplions le coût par œuf par le nombre d'œufs dans la boîte.

$$
\text{Coût par boîte de 6 œufs} = \text{Coût par œuf} \times 6
$$

En remplaçant les valeurs :

$$
\text{Coût par boîte de 6 œufs} = 0,3 \, \text{€} \times 6 \approx 1,8 \, \text{€}
$$

##### Résultat Final

Ainsi, le coût de production pour une boîte d'œufs (avec 6 œufs) serait **environ 1,80 euros**.

#### Calcul du Prix de Vente de la Viande de Pintade

Pour déterminer le prix de vente de la viande de pintade que nous allons proposer, voici les étapes que nous avons suivies :

##### 1. Déterminer le poids de la viande après l'abattage

La pintade entière a un poids moyen d'environ 1,65 kg. Après l'abattage, nous pouvons obtenir environ 70 % du poids total en viande, en tenant compte des viscères et d'autres parties non comestibles.

$$
\text{Poids de la viande} = \text{Poids total} \times 0,70 = 1,65 \, \text{kg} \times 0,70 \approx 1,155 \, \text{kg}
$$

##### 2. Prix par kilogramme de viande

Si le prix de vente au consommateur final est de 14,00 €/kg pour la pintade entière, nous supposons que le prix par kilogramme de viande est dans la même gamme.

##### 3. Calcul du prix de vente de la viande

Le prix de vente de la viande serait alors :

$$
\text{Prix de vente de la viande} = \text{Prix par kg} \times \text{Poids de la viande}
$$

En remplaçant :

$$
\text{Prix de vente de la viande} = 14,00 \, \text{€/kg} \times 1,155 \, \text{kg} \approx 16,17 \, \text{€}
$$

##### Résumé

Par conséquent, nous pourrions vendre la viande de pintade pour environ **16,17 euros**.

#### Abattage de la Pintade

Nous avons décidé d'abattre la pintade que nous avons élevée. Pour cela, voici le coût que nous devons considérer :

1. **Coût de l'abattage** :  
   Le coût de l'abattage par pintade est d'environ **5,00 €**.

##### Conclusion

Ainsi, le coût total pour abattre une pintade serait d'environ **5,00 euros**.


In [67]:
cf_oeuf = 1.17 
cf_boite_oeuf = 7
cout_boite_oeuf = 1.8
cf_pintada = 16.17
cout_abattage_pintade = 5
print("Vente d'une boîte de 6 œufs: ", cf_boite_oeuf, "€")
print("Vente de viande de pintade: ", cf_pintada, "€")
print("Coût de production d'une boîte de 6 œufs: ", cout_boite_oeuf, "€")
print("Coût de l'abattage de la pintade: ", cout_abattage_pintade, "€")

Vente d'une boîte de 6 œufs:  7 €
Vente de viande de pintade:  16.17 €
Coût de production d'une boîte de 6 œufs:  1.8 €
Coût de l'abattage de la pintade:  5 €


<font color = green>
Partie 4 : Coût du transport <br/> 
</font>

#### 1. Coûts d'acquisition des véhicules
Il faut inclure le coût d'achat initial de chaque type de véhicule :

**Camionnette** : entre 25 000 € et 45 000

**Fourgonnette** : entre 30 000 € et 50 000*

**Estafette** : entre 25 000 € et 45 000


In [68]:
prix_camionnette = int((25000 + 45000)/2)
prix_fourgonnette = int((30000 + 50000)/2)
prix_estafette = int((25000 + 45000)/2)
print("Prix Moyen d'une Camionnette: ", prix_camionnette, "€")
print("Prix Moyen d'une Fourgonnette: ", prix_fourgonnette, "€")
print("Prix Moyen d'une Estafette: ", prix_estafette, "€")

Prix Moyen d'une Camionnette:  35000 €
Prix Moyen d'une Fourgonnette:  40000 €
Prix Moyen d'une Estafette:  35000 €


#### 2. Coûts d'exploitation
Pour chaque véhicule, nous calculons les frais de fonctionnement mensuels :

##### Assurance

**Camionnette :** Assurance annuelle : 600 € à 1 200 € 

**Fourgonnette :** Assurance annuelle : 800 € à 1 500 €. 

**Estafette :** Assurance annuelle : 1 000 € à 1 800 €. 

In [69]:
assurance_c = int((600 + 1200)/2)
assurance_f = int((800 + 1500)/2)
assurance_e = int((1000 + 1800)/2)
print(" Assurance anunuelle pour la Camionnette: ", assurance_c, "€")
print(" Assurance anunuelle pour la Fourgonnette: ", assurance_f, "€")
print(" Assurance anunuelle pour la Estafette: ", assurance_e, "€")

 Assurance anunuelle pour la Camionnette:  900 €
 Assurance anunuelle pour la Fourgonnette:  1150 €
 Assurance anunuelle pour la Estafette:  1400 €


##### Entretien

**Camionnette :** Coût d'entretien annuel moyen : 500 € à 1 000 €.

**Fourgonnette :** Coût d'entretien annuel moyen : 800 € à 1 500 €.

**Estafette :** Coût d'entretien annuel moyen : 1 200 à 2 000 euros.

In [70]:
entretien_c = int((500 + 1000)/2)
entretien_f = int((800 + 1500)/2)
entretien_e = int((1200 + 2000)/2)
print(" Entretien anunuelle pour la Camionnette: ", entretien_c, "€")
print(" Entretien anunuelle pour la Fourgonnette: ", entretien_f, "€")
print(" Entretien anunuelle pour la Estafette: ", entretien_e, "€")

 Entretien anunuelle pour la Camionnette:  750 €
 Entretien anunuelle pour la Fourgonnette:  1150 €
 Entretien anunuelle pour la Estafette:  1600 €


#### 3. Capacité de Transport : Viande et Œufs de Pintade

Pour estimer la capacité de transport en termes de viande et d'œufs de pintade, nous avons pris en compte les éléments suivants :

**Poids moyen de la viande de pintade** : 1,155 kg (prête à la vente).

**Poids d'une caisse d'œufs** : environ 0,250 kg pour 6 œufs.

La capacité de transport des véhicules en termes de volume et de poids est la suivante :

**Camionnette** : Capacité de 3 à 4m³, ce qui correspond à une charge d'environ 500 kg à 800 kg.

**Fourgonnette** : Capacité de 5 à 6m³, correspondant à une charge de 800 kg à 1.200 kg.

**Estafette** : Capacité de 7 à 9m³, permettant de transporter jusqu'à 1.500 kg de charge.



In [71]:
capacite_c = int((500 + 800)/2)
capacite_f = int((800 + 1200)/2)
capacite_e = 1500
print(" Capacité de la Camionnette: ", capacite_c , "kg")
print(" Capacité de la Fourgonnette: ", capacite_f, "kg")
print(" Capacité de la Estafette: ", capacite_e, "kg")

 Capacité de la Camionnette:  650 kg
 Capacité de la Fourgonnette:  1000 kg
 Capacité de la Estafette:  1500 kg


In [72]:
poid_pintada = 1
capacite_viande_c = int(capacite_c/1.555)
capacite_boite_oeufs_c = int(capacite_c/0.250)
capacite_viande_f = int(capacite_f/1.555)
capacite_boite_oeufs_f = int(capacite_f/0.250)
capacite_viande_e = int(capacite_e/1.555)
capacite_boite_oeufs_e = int(capacite_e/0.250)
print(" Capacité de la Camionnette: ", capacite_viande_c , "pintades ou ", capacite_boite_oeufs_c, "boîte d'œufs")
print(" Capacité de la Fourgonnette: ", capacite_viande_f , "pintades ou ", capacite_boite_oeufs_f, "boîte d'œufs")
print(" Capacité de la Estafette: ", capacite_viande_e , "pintades ou ", capacite_boite_oeufs_e, "boîte d'œufs")


 Capacité de la Camionnette:  418 pintades ou  2600 boîte d'œufs
 Capacité de la Fourgonnette:  643 pintades ou  4000 boîte d'œufs
 Capacité de la Estafette:  964 pintades ou  6000 boîte d'œufs


#### 4. Distance et Calcul de l'Essence

Nous avons estimé que la ferme se trouve à environ 50 km du port de Marseille, ce qui implique un aller-retour de 100 km.

Le calcul du coût de l'essence dépend du type de véhicule utilisé. Nous avons pris en compte les données suivantes :

**Prix de l'essence à Marseille** : environ 1,90€/litre.

##### Consommation moyenne :

**Camionnette** : environ 8-10 km/litre.

**Fourgonnette** : environ 6-8 km/litre.

**Estafette** : environ 5-6 km/litre.

En fonction de la consommation moyenne et de la distance à parcourir (100 km aller-retour), voici le coût de l'essence pour chaque véhicule :

In [73]:
prix_essence = 1.9 
essence_c = (100/((8 + 10)/2))*prix_essence
essence_f = (100/((6 + 8)/2))*prix_essence
essence_e = (100/((5 + 6)/2))*prix_essence
print("Prix d'un trajet de 100 km:")
print("Camionnette: ", essence_c, "€")
print("Fourgonnette: ", essence_f, "€")
print("Estafette: ", essence_e, "€")


Prix d'un trajet de 100 km:
Camionnette:  21.11111111111111 €
Fourgonnette:  27.142857142857142 €
Estafette:  34.54545454545455 €


<font color = green>
Partie 5 : Coûts de Maintenance de la Ferme à Marseille <br/> 
</font>

**1. Impôts et Taxes**

Impôt Foncier (Taxe Foncière) :

Pour les propriétés rurales à Marseille, l'impôt peut varier. Une estimation raisonnable serait de 500€ à 1.200€ par an.

**2. Assurance de la Propriété**

Assurance Habitation et Propriété Rurale :

L'assurance dans la région peut varier, mais une moyenne raisonnable serait de 400€ à 800€ par an.

**3. Entretien de la Propriété**

Entretien des Bâtiments et Clôtures :

Les coûts d'entretien peuvent se situer entre 150€ et 350€ par mois à Marseille, en raison du climat et de la nécessité d'un entretien régulier.

Eau et Électricité :

En fonction de l'utilisation, les coûts des services publics peuvent être de 80€ à 200€ par mois.


**4. Autres Coûts**

Dépenses Générales et Imprévus :

Un montant raisonnable pour les dépenses inattendues peut être de 100€ à 200€ par mois.

In [74]:
impot = int((500 + 1200)/2)
assurance = int((400+800)/2)
entretien = int((150+350)/2)*12
eau_electricite = int((80+200)/2)*12
autres_couts = int((100+200)/2)*12
cout_an = impot + assurance + entretien + eau_electricite + autres_couts
print("Coût annuel de maintien de la ferme: ", cout_an, "€")

Coût annuel de maintien de la ferme:  7930 €


<font color = blue>
Résolution du problème <br/>
</font>


<font color = green>
Partie 1:  <br/>
</font>


Après 2 ans, nous avons 882 pintades, la ferme a une capacité maximale de 900 pintades et nous considérons dans notre année que les 6 premiers mois les pintades ne sont pas en période de reproduction, sur cette base le profit des 6 premiers mois sera basé uniquement sur la vente de viande, en plus de ce facteur nous devons considérer que nous avons besoin de 3 femelles pour 1 mâle pour la reproduction. 


In [75]:
femmes_machos = 3/4

Six premiers mois: Sachant que nous allons vendre les anciennes pintades et que nous avons beaucoup de mâles.

In [76]:
vendre_p11 = int(pintadas_1 + (pintadas_2 - pintadas_femmes_1)*(1 - femmes_machos))
print("Nombre minimum de pintadas que nous pouvons vendre: ", vendre_p11, "pintades")
cout_vp11 = cout_abattage_pintade*vendre_p11
print("Coût de l'abattage de ces pintades: ", cout_vp11, "€")
cf_vp11 = cf_pintada*vendre_p11
print("Le produit de la vente de pintades: ", cf_vp11, "€")


Nombre minimum de pintadas que nous pouvons vendre:  257 pintades
Coût de l'abattage de ces pintades:  1285 €
Le produit de la vente de pintades:  4155.6900000000005 €


Comme tous les moyens de transport supportent au moins 400 pintades, nous allons toutes les vendre d'un coup pour minimiser les coûts de transport.

In [77]:
cout_vp11_c = cout_vp11 + essence_c 
cout_vp11_f = cout_vp11 + essence_f
cout_vp11_e = cout_vp11 + essence_e
profit_vp11_c = cf_vp11 - cout_vp11_c
profit_vp11_f = cf_vp11 - cout_vp11_f
profit_vp11_e = cf_vp11 - cout_vp11_e
print("Le profit pour la: ")
print("Camionnette :", profit_vp11_c, "€")
print("Fourgonnette :", profit_vp11_f, "€")
print("Estafette :", profit_vp11_e, "€")


Le profit pour la: 
Camionnette : 2849.578888888889 €
Fourgonnette : 2843.5471428571436 €
Estafette : 2836.144545454546 €


Après, il reste 469 femelles et 156 males.(625 en totale)

En considerant que le pintades ne produisent rien avant 6 mois, donc pendant la période de couvaison de la troisième année, que ces 469 femelles produisent œufs.Il n'y a aucun sens à incuber des œufs dès le début.

Selon le demande de marché, il faut aller au vieux-port pour vendre les œufs chaque mois.(Il faut vendre les œufs frais.)

Selon le modélisation préceédent, une pintade peut produire 10 œufs chaque mois

In [78]:
pintades_3 = pintadas_2 - vendre_p11
pintade_femme_3 = int(femmes_machos*pintades_3)
egg_chaque_mois = int(pintade_femme_3*10)
boite_chaque_mois1 = int(egg_chaque_mois/6)
print("Le nombre maximum de oeufs produit chaqun mois:", egg_chaque_mois, "oeufs")
print("Nombre de boîte de oeufs",boite_chaque_mois1, "boîtes" )

Le nombre maximum de oeufs produit chaqun mois: 4680 oeufs
Nombre de boîte de oeufs 780 boîtes


Pour réduire les coûts liés à l'entretien des pintades à la ferme, nous allons vendre tous les œufs pendant les cinq premiers mois et en garder quelques-uns le dernier mois, afin d'avoir de nouvelles pintades pour l'année prochaine.

In [79]:
vendre_o11 = cf_boite_oeuf*boite_chaque_mois1
profit_vo11_c = vendre_o11 - essence_c - boite_chaque_mois1*cout_boite_oeuf
profit_vo11_f = vendre_o11 - essence_f - boite_chaque_mois1*cout_boite_oeuf
profit_vo11_e = vendre_o11 - essence_e - boite_chaque_mois1*cout_boite_oeuf
print("Le profit pour la: ")
print("Camionnette :", profit_vo11_c, "€")
print("Fourgonnette :", profit_vo11_f, "€")
print("Estafette :", profit_vo11_e, "€")

Le profit pour la: 
Camionnette : 4034.8888888888887 €
Fourgonnette : 4028.857142857143 €
Estafette : 4021.454545454545 €


Après six mois, nous devons garder au moins 675 femelles pour l'année prochaine et 225 mâles. Pour cela, nous devrons conserver un minimum de 1 688 œufs afin de garantir le nombre de femelles.

In [80]:
vendre_o12 = int((egg_chaque_mois - 1692)/6)*(cf_boite_oeuf - cout_boite_oeuf)
profit_vo12_c = vendre_o12 - essence_c
profit_vo12_f = vendre_o12 - essence_f
profit_vo12_e = vendre_o12 - essence_e
print("Le profit pour la: ")
print("Camionnette :", profit_vo12_c, "€")
print("Fourgonnette :", profit_vo12_f, "€")
print("Estafette :", profit_vo12_e, "€")

Le profit pour la: 
Camionnette : 2568.4888888888886 €
Fourgonnette : 2562.4571428571426 €
Estafette : 2555.0545454545454 €


Avec cela, le profit attendu pour la troisième année, sans l'incubation, serait de: 

In [81]:
profit3_c = profit_vp11_c + profit_vo11_c*5 + profit_vo12_c - cout_an - assurance_c - entretien_c
profit3_f = profit_vp11_f + profit_vo11_f*5 + profit_vo12_f - cout_an - assurance_f - entretien_f
profit3_e = profit_vp11_f + profit_vo11_f*5 + profit_vo12_f - cout_an - assurance_e - entretien_e
print("Le profit pour la: ")
print("Camionnette :", profit3_c, "€")
print("Fourgonnette :", profit3_f, "€")
print("Estafette :", profit3_e, "€")

Le profit pour la: 
Camionnette : 16012.512222222223 €
Fourgonnette : 15320.290000000005 €
Estafette : 14620.290000000005 €



Après ces 6 mois, Il faut vendre tous ces 625 pintades parce qu'ils ne produire rien.

Donc nous pouvons déduire le meilleur résolution:

Nous ne vendons que les oeufs dans ces 6 mois. Dans le dernier mois, nous gardons 900 oeufs et faire l'inbucation artificielle. Et nous vendons tous les pintades.


In [82]:
egg_profit = (egg_chaque_mois * 6 - 900)  * (cf_boite_oeuf-cout_boite_oeuf)/6 
pintade_profit = pintade_totale * (cf_pintada - cout_abattage_pintade)
cost_inbuction = 900*2
print("profit d'oeufs:", egg_profit)
print("profit des pintades:", pintade_profit)
print("cost d'inbuction:", cost_inbuction)

profit d'oeufs: 23556.0
profit des pintades: 6981.250000000001
cost d'inbuction: 1800


Choix de voiture:
Pour le premier 5 mois, nous transportons 15008 oeufs chaqun mois, c'est à dire que 2502 boite d'oeufs chaque mois.Tous ces 3 voitures peuvent le transporter une fois.

Pour le dernier mois, nous transportons 625 pintades et 14108 oeufs(2351 boite). 

Pour le cammionette, il faut 8 trajets.
Pour le fourgonnette, il faut 7 trajets.
Pour l'estafette, il faut 6 trajets.

Nous allons comparer les costs.

In [83]:
cost_transport_c = essence_c * 8 + prix_camionnette + assurance_c + entretien_c
cost_transport_f = essence_f * 7 + prix_fourgonnette + assurance_f + entretien_f
cost_transport_e = essence_e * 6 + prix_estafette + assurance_e + entretien_e
print("cost de cammionette:", cost_transport_c)
print("cost de fourgonnette:", cost_transport_f)
print("cost d'estafette:", cost_transport_e)

cost de cammionette: 36818.88888888889
cost de fourgonnette: 42490.0
cost d'estafette: 38207.27272727273


Selon le résultats précédent, nous devons choisir une cammionette ou un estafette. En tenant compte d'une éventuelle expansion de la ferme à l'avenir, nous choissions un estafette.

Nous pouvons déduire le profit jusqu'à la fin de troisième année.

In [84]:
Profit_totale = egg_profit + pintade_profit + cf_vp11 - cout_vp11 - cost_transport_e - cout_an*3 - cost_inbuction
print("Profit jusqu'à la fin de trosième année:",Profit_totale)

Profit jusqu'à la fin de trosième année: -30389.332727272726
